presto-hive-hadoop2/src/main/java/com/facebook/presto/hive/HiveHadoop2Plugin.java

```diff
@@ -13,11 +13,21 @@
  */
 package com.facebook.presto.hive;
 
+import com.facebook.presto.hive.metastore.ExtendedHiveMetastore;
+import com.google.inject.Module;
+
+import java.util.Optional;
+
 public class HiveHadoop2Plugin
         extends HivePlugin
 {
     public HiveHadoop2Plugin()
     {
         super("hive-hadoop2");
     }
+
+    public HiveHadoop2Plugin(String name, Optional<ExtendedHiveMetastore> metastore, Module pathProviderModule)
```

Comment:
Why does the `PathFilterProvider` has to be wrapped in a module? Why not simply supply the `PathFilterProvider` itself?

e.g.: change the interface to `public HiveHadoop2Plugin(String name, Optional<PathFilterProvider> pathFilterProvider)`

Reply:
Done!